code/modules/admin/verbs/debug.dm

```diff
@@ -120,52 +74,75 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
-		var/returnval = null
+	callproc(usr,target)
+	feedback_add_details("admin_verb","APC") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
 
-		var/procname = input("Proc path, eg: /proc/fake_blood","Path:", null) as text|null
-		if(!procname)
-			return
+/client/proc/callproc(mob/user, var/datum/target as anything)
+	if(!check_rights(R_DEBUG))
+		return
 
-		if(!hascall(target, procname))
-			to_chat(usr, "<span class='red'>Error: callatomproc(): target has no such call [procname].</span>")
-			return
+	var/lst[] // List reference
+	lst = new/list() // Make the list
+	var/targetselected = !isnull(target)
 
-		var/argnum = input("Number of arguments","Number:",0) as num|null
-		if(!argnum && (argnum!=0))
-			return
+	var/procname = input("Proc path, eg: /proc/fake_blood","Path:", null) as text|null
+	if(!procname)
+		return
 
-		lst.len = argnum // Expand to right length
-		//TODO: make a list to store whether each argument was initialised as null.
-		//Reason: So we can abort the proccall if say, one of our arguments was a mob which no longer exists
-		//this will protect us from a fair few errors ~Carn
+	// Do not make this a global reference. Global references can be cleared out.
+	if (istype(target, /datum/subsystem/dbcore/))
+		to_chat(user, "<span class='red'>Never use atom proc call to inject SQL.</span>")
+		message_admins("[key_name(user)] used atom proc call on the db controller.")
+		log_admin("[key_name(user)] used atom proc call on the db controller.")
+		return
 
-		var/i
-		for(i = 1, i < argnum + 1, i++) // Lists indexed from 1 forwards in byond
-			lst[i] = variable_set(src)
+	if(target && !hascall(target, procname))
+		to_chat(user, "<span class='red'>Error: calladvproc(): target has no such call [procname].</span>")
+		return
 
-		log_admin("[key_name(src)] called [target]'s [procname]() with [lst.len ? "the arguments [list2params(lst)]":"no arguments"].")
-		returnval = call(target,procname)(arglist(lst)) // Pass the lst as an argument list to the proc
+	var/argnum = input("Number of arguments","Number:",0) as num|null
+	if(!argnum && (argnum!=0))
+		return
 
-		if(isnull(returnval))
-			returnval = "null"
-		else if(returnval == "")
-			returnval = "\"\" (empty string)"
+	lst.len = argnum // Expand to right length
+	//TODO: make a list to store whether each argument was initialised as null.
+	//Reason: So we can abort the proccall if say, one of our arguments was a mob which no longer exists
+	//this will protect us from a fair few errors ~Carn
 
-		var/returntext = returnval
-		if(istype(returnval, /datum))
-			returntext = "[returnval] <a href='?_src_=vars;Vars=\ref[returnval]'>\[VV\]</A>"
-			spawn(PROC_RESULT_KEEP_TIME)
-				returnval = null
-		else if(istype(returnval, /list))
-			returntext = "<a href='?_src_=vars;List=\ref[returnval]'>\[List\]</A>"
-			spawn(PROC_RESULT_KEEP_TIME)
-				returnval = null
+	var/i
+	for(i = 1, i < argnum + 1, i++) // Lists indexed from 1 forwards in byond
```

Comment:
```suggestion
	for(var/i = 1, i < argnum + 1, i++)
```
Unnecessary comment as anyone digging through code should already know this

Reply:
i didn't write it, will remove